bobtask/input.go

```diff
@@ -61,13 +65,42 @@ func (t *Task) FilteredInputs(projectRoot string) (_ []string, err error) {
 				continue
 			}
 			inputDirtyRooted[i] = filepath.Join(t.dir, input)
+
+			for _, prefix := range []string{goinputdiscovery.Keyword} {
+				if strings.HasPrefix(input, prefix) {
+					inputDirtyRooted[i] = prefix + filepath.Join(t.dir, strings.TrimPrefix(input, prefix))
+					continue outerLoop
+				}
+			}
+		}
+	}
+
+	// Auto input discovery
+	var inputDirtyRootedDiscovered []string
+	for _, input := range inputDirtyRooted {
+		// TODO: collect keywords for auto discovery some place
+		if strings.HasPrefix(input, goinputdiscovery.Keyword+inputdiscovery.KeywordSeparator) {
+			packagePathRel := filepath.Clean(strings.TrimPrefix(input, goinputdiscovery.Keyword+inputdiscovery.KeywordSeparator))
```

Comment:
Is `Clean()` required?

Reply:
no, I just looked into the code of `filepath.Join`. It will do a `Clean` after it joins the elements.